File: ckgame-java/src/main/java/com/ckgame/core/TitleTier.java
package com.ckgame.core;

public enum TitleTier {
    BARONY(1, "男爵"),
    COUNTY(2, "伯爵"),
    DUCHY(3, "公爵"),
    KINGDOM(4, "国王"),
    EMPIRE(5, "皇帝");

    private final int value;
    private final String rankName;

    TitleTier(int value, String rankName) {
        this.value = value;
        this.rankName = rankName;
    }

    public int value() { return value; }

    public String rankName() { return rankName; }

    public double creationCost() {
        return switch (this) {
            case BARONY -> 0.0;
            case COUNTY -> 50.0;
            case DUCHY -> 200.0;
            case KINGDOM -> 500.0;
            case EMPIRE -> 1000.0;
        };
    }

    public boolean isDestroyable() {
        return this.compareTo(DUCHY) >= 0;
    }
}

File: ckgame-rust/src/core/title_tier.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum TitleTier {
    Barony = 1,
    County = 2,
    Duchy = 3,
    Kingdom = 4,
    Empire = 5,
}

impl TitleTier {
    pub fn rank_name(&self) -> &'static str {
        match self {
            TitleTier::Barony => "男爵",
            TitleTier::County => "伯爵",
            TitleTier::Duchy => "公爵",
            TitleTier::Kingdom => "国王",
            TitleTier::Empire => "皇帝",
        }
    }

    pub fn tier_prefix(&self) -> &'static str {
        match self {
            TitleTier::Barony => "b",
            TitleTier::County => "c",
            TitleTier::Duchy => "d",
            TitleTier::Kingdom => "k",
            TitleTier::Empire => "e",
        }
    }

    pub fn creation_cost(&self) -> f64 {
        match self {
            TitleTier::Barony => 0.0,
            TitleTier::County => 50.0,
            TitleTier::Duchy => 200.0,
            TitleTier::Kingdom => 500.0,
            TitleTier::Empire => 1000.0,
        }
    }

    pub fn is_destroyable(&self) -> bool {
        (*self as u8) >= (TitleTier::Duchy as u8)
    }
}